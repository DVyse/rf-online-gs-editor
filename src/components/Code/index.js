/**
 *
 * Code
 *
 */

import { Label } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

class Code extends React.PureComponent {
  static propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func,
  };

  constructor(props) {
    super(props);

    this.createRef = this.createRef.bind(this);
    this.onClickSelect = this.onClickSelect.bind(this);
  }

  createRef(target) {
    this.elementRef = target;
  }

  onClickSelect(evt) {
    evt.preventDefault();
    const range = document.createRange();
    const sel = window.getSelection();
    range.selectNodeContents(this.elementRef);
    sel.removeAllRanges();
    sel.addRange(range);
  }

  render() {
    const { children, onClick } = this.props;

    return (
      <Label>
        <CodeStyled
          ref={this.createRef}
          onClick={onClick || this.onClickSelect}
        >
          {children}
        </CodeStyled>
      </Label>
    );
  }
}

export default Code;

const CodeStyled = styled.code`
  color: rgb(224, 57, 151);
`;
