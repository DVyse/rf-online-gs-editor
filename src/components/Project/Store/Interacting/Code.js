/**
 *
 * ProjectStoreInteractingCode
 *
 */

import { Input } from 'semantic-ui-react';
import { Map } from 'immutable';

import { getCode } from '~/containers/App/getters/projectStore';
import PropTypes from 'prop-types';
import React from 'react';

/* eslint-disable react/prefer-stateless-function */
class ProjectStoreInteractingCode extends React.PureComponent {
  constructor(props) {
    super(props);

    this.changeValue = evt => {
      const { onChangeValue, store } = this.props;
      onChangeValue(store, evt.target.value);
    };
  }

  render() {
    const {
      store,
      storeNextValues,
      size,
      className,
      label,
      fluid,
    } = this.props;

    const value = getCode(storeNextValues.get('nextValue'), {
      entry: store,
    });

    return (
      <Input
        size={size}
        fluid={fluid}
        type="text"
        value={value}
        onChange={this.changeValue}
        className={className}
        label={label}
      />
    );
  }
}

ProjectStoreInteractingCode.propTypes = {
  store: PropTypes.instanceOf(Map).isRequired,
  storeNextValues: PropTypes.instanceOf(Map).isRequired,
  onChangeValue: PropTypes.func.isRequired,
  size: PropTypes.oneOf(['mini', 'small', 'large', 'big', 'huge', 'massive']),
  className: PropTypes.string,
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.number,
  ]),
  fluid: PropTypes.bool,
};

ProjectStoreInteractingCode.defaultProps = {
  size: 'mini',
  className: '',
  label: null,
  fluid: true,
};

export default ProjectStoreInteractingCode;
