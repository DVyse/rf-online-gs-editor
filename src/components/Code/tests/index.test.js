import React from 'react';
import { shallow } from 'enzyme';

import Code from '../index';

describe('<Code />', () => {
  it('Should exist', () => {
    const wrapper = shallow(<Code />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Code__Text').prop('onClick')).toBeInstanceOf(Function);
  });
  it('Should have children', () => {
    const children = 'children';
    const wrapper = shallow(<Code>{children}</Code>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Code__Text').text()).toEqual(children);
  });
  it('Should have onClick', () => {
    const onClick = () => null;
    const wrapper = shallow(<Code onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Code__Text').prop('onClick')).toBeInstanceOf(Function);
  });
});
