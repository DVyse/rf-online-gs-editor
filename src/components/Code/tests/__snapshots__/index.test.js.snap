// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> Should exist 1`] = `
<Label>
  <Code__Text
    onClick={[Function]}
  />
</Label>
`;

exports[`<Code /> Should have children 1`] = `
<Label>
  <Code__Text
    onClick={[Function]}
  >
    children
  </Code__Text>
</Label>
`;

exports[`<Code /> Should have onClick 1`] = `
<Label>
  <Code__Text
    onClick={[Function]}
  />
</Label>
`;

exports[`<Code /> Should work without errors 1`] = `
<Label>
  <Code__Text
    onClick={[Function]}
  />
</Label>
`;
